docs/website/docs/guides/ml-frameworks/pytorch.md

```diff
@@ -322,6 +322,132 @@ their values independently at runtime.
         self.value = new_value
     ```
 
+#### :octicons-code-16: Separating parameters during compilation
+
+In practical scenarios, it is usually unfeasible to embed model parameters
+in the IR.
+
+`aot.externalize_module_parameters()` allows for the separation of
+the parameters from the IR, but still encodes a symbolic relationship between
+between parameters and the IR, so they can be loaded at runtime.
+
+We define a `torch.nn.Module` that will be compiled, and save
+random tensors to be loaded as parameters in a safetensors file.
+
+```python
+import torch
+class LinearModule(torch.nn.Module):
+    def __init__(self, in_features, out_features):
+        super().__init__()
+        self.weight = torch.nn.Parameter(torch.randn(in_features, out_features))
+        self.bias = torch.nn.Parameter(torch.randn(out_features))
+
+    def forward(self, input):
+        return (input @ self.weight) + self.bias
+
+# Construct LinearModule.
+linear_module = LinearModule(4,3)
+```
+
+Safetensors can be created directly from a parameters dictionary created from
+randomly generated torch tensors.
+
+```python
+# Create a params dictionary. Note that the keys here match
+# LinearModule's attributes. We will use the saved safetensor
+# file for use from the command line.
+from safetensors.torch import save_file
+
+wt = torch.randn(4,3)
+bias = torch.randn(3)
+params = { "weight": wt, "bias": bias }
+save_file(params, "params.safetensors")
+```
+
+Exporting through `aot.export()` and compiling the produced output generates a binary.
+
+```python
+import numpy as np
+import shark_turbine.aot as aot
+
+# Externalize the model parameters. This removes weight
+# tensors from the IR module, allowing them to be loaded
+# at runtime. Symbolic references to these parameters are
+# still retained in the IR.
+aot.externalize_module_parameters(linear_module)
+
+ext_arg = torch.randn(4)
+exported_module = aot.export(linear_module, ext_arg)
+
+# Compile the exported module, to generate the binary.
+binary = exported_module.compile(save_to=None)
+
+# Save the input as an npy tensor, for later reuse.
+ext_arg_2 = ext_arg.numpy()
+np.save("input.npy", ext_arg_2)
```

Comment:
Ah, `ext_arg` is an input! I'd put a comment before you define `ext_arg` above.

Reply:
maybe rename ext_arg entirely?

Reply:
Ok, will do.